### ISSUE TITLE
Loss is very high please help

### ISSUE THREAD

**Issue by krist929** [canopyai/Orpheus-TTS-0.1#286]
<img width="1109" height="657" alt="Image" src="https://github.com/user-attachments/assets/9a264d97-1b4a-4163-9dcf-7d09770962e2" />

Hello everyone I am attempting to full fine tune and I cant seem to get my loss rate below 6-7.  This is the dataset https://huggingface.co/datasets/Kriest/firefly - raw files
https://huggingface.co/datasets/Kriest/firefly-tokenized - tokenized (potentially wrong)

I have messed a lot with the learning rate and all sorts but this is my current setup.  

TTS_dataset: Kriest/firefly-tokenized

model_name: "canopylabs/orpheus-tts-0.1-pretrained"

# Training Args
epochs: 7
batch_size: 2
number_processes: 1
pad_token: 128263
save_steps: 10000  
learning_rate: 1e-5
warmup_steps: 50
weight_decay: 0.0

# Naming and paths
save_folder: "checkpoints"
project_name: "tuning-orpheus"
run_name: "bf16-1e-5-7epochs"

training_args = TrainingArguments(
    overwrite_output_dir=True,
    num_train_epochs=epochs,
    per_device_train_batch_size=batch_size, 
    logging_steps=1,
    bf16=True,  # Using bf16 for training
    fp16=False,  # Ensure fp16 is disabled
    output_dir=f"./{base_repo_id}",
    report_to="wandb", 
    save_steps=save_steps,
    remove_unused_columns=False,
    learning_rate=learning_rate,
    warmup_steps=config.get("warmup_steps", 0),
    lr_scheduler_type="constant_with_warmup",  # Prevent LR decay
    weight_decay=weight_decay,  # Regularization
    gradient_accumulation_steps=4,
)

Any and all help is greatly appreciated.  If it takes me some time to respond to test suggestions its because  I am waiting on someone to hop off the GPU rental I am using :)

**Comment by shekharmeena2896** [comment]
1. Firstly full fine tuning is not work with your dataset because the size of the data is so low lesss than 500 rows and maybe less than hour for full finetuning you would require hundreds of hour of data and you will need to freeze certain layers.
2. secondly with your amount of data on finetune is possible, if your dataset is single speaker, this will do the finetune for single speaker only.

**Comment by krist929** [comment]
> 1. Firstly full fine tuning is not work with your dataset because the size of the data is so low lesss than 500 rows and maybe less than hour for full finetuning you would require hundreds of hour of data and you will need to freeze certain layers.
> 2. secondly with your amount of data on finetune is possible, if your dataset is single speaker, this will do the finetune for single speaker only.



That was the plan to fine tune it on the one voice so it sounds like that person. If I don't have enough data for the fine tune would peft/lora be the way to go? I was trying to get higher quality because I am trying to plug the voice into an book to audiobook project a friend is working on and I am supposed to be trying TTS models to see which sounds the best. 

Forgive me if I misunderstand still learning - also if this is not how you reply I don't use GitHub much

**Comment by shekharmeena2896** [comment]
you should use lora, the unsloth finetuning script is provided in the github of this repo, simply paste your data and finetune it, follow the instructions  https://colab.research.google.com/github/unslothai/notebooks/blob/main/nb/Orpheus_(3B)-TTS.ipynb


make sure you have good gpu like a100 or something, if not than you can finetune it using lesser steps in t4 gpu.

**Comment by krist929** [comment]
> you should use lora, the unsloth finetuning script is provided in the github of this repo, simply paste your data and finetune it, follow the instructions https://colab.research.google.com/github/unslothai/notebooks/blob/main/nb/Orpheus_(3B)-TTS.ipynb
> 
> make sure you have good gpu like a100 or something, if not than you can finetune it using lesser steps in t4 gpu.

Got it I'll give that a try.  Thanks for the help
